09-features.md

```diff
@@ -42,7 +42,9 @@ The Context column decodes as follows:
 | 22/23 | `option_anchors_zero_fee_htlc_tx` | Anchor commitment type with zero fee HTLC transactions   | IN       | `option_static_remotekey` | [BOLT #3][bolt03-htlc-tx], [lightning-dev][ml-sighash-single-harmful]|
 | 26/27 | `option_shutdown_anysegwit`         | Future segwit versions allowed in `shutdown`              | IN       |                   | [BOLT #2][bolt02-shutdown]   |
 | 44/45 | `option_channel_type`            | Node supports the `channel_type` field in open/accept     | IN       |                   | [BOLT #2](02-peer-protocol.md#the-open_channel-message) |
+| 46/47 | `option_scid_alias`              | Supply channel aliases for routing                        | IN       |                   | [BOLT #2][bolt02-channel-ready]   |
 | 48/49 | `option_payment_metadata` | Payment metadata in tlv record | 9 | | [BOLT #11](11-payment-encoding.md#tagged-fields)
+| 50/51 | `option_zeroconf`                | Understands zeroconf channel types                        | IN       | `option_scid_alias` | [BOLT #2][bolt02-channel-ready]   |
```

Comment:
If there is no dependency link, how do you enforce aliases are used for zero-conf channels? Is this not the point of option-scid-alias feature bit?

Reply:
The dependency here only applies to the init and node_announcement contexts. There is no such dependency in the channel type contexts. Seems to me like it makes perfect sense to have an scid-alias dependency in those contexts.

Reply:
Assuming that we rename `option_scid_alias` to `option_scid_privacy` (as you suggested and I agree with you).

The two features introduced by this PR: `option_scid_privacy` and `option_zeroconf`, are very different and independent from each other. We can have one without the other. It just happens that they both depend on supporting scid alias, but there is no explicit feature for supporting aliases, just like e.g. `option_payment_metadata` depends on supporting tlvs.

> The dependency here only applies to the init and node_announcement contexts. There is no such dependency in the channel type contexts

Dependencies as layed out in BOLT 9 are not context dependent though.

Reply:
~How can we ever announce a channel, if setting the `option_scid_alias` literally forces us to set the `announce_channel` flag in the `open_channel` message, and that flag is impossible to renegotiate once the negotiation is done? I think this literally makes public/announced channels and zeroconf mutually exclusive.~

Oh I see, it can still be negotiated as a blank feature, not as part of the `channel_type` at all.

https://github.com/lightning/bolts/pull/910/files#diff-ed04ca2c673fd6aabde69389511fa9ee60cb44d6b2ef6c88b549ffaa753d6afeR246-R247